backend/src/main/kotlin/com/tacs/backend/service/ChampionshipService.kt

```
@@ -0,0 +1,131 @@
+import com.tacs.backend.DAO.UserByChampionshipDAO
+import com.tacs.backend.DAO.UserDAO
+import com.tacs.backend.entity.Championship
+import com.tacs.backend.entity.User
+import com.tacs.backend.entity.UserByChampionship
+import com.tacs.backend.entity.UserByChampionshipId
+import com.tacs.backend.exception.ChampionshipNotFoundException
+import com.tacs.backend.exception.UnknownUserException
+import com.tacs.backend.request.CreateChampionshipRequest
+import com.tacs.backend.request.VisibilityType
+import com.tacs.backend.response.ChampionshipResponse
+import com.tacs.backend.response.ChampionshipScoreResponse
+import com.tacs.backend.response.ScoreByUser
+import org.springframework.stereotype.Component
+import java.util.*
+
+@Component
+class ChampionshipService(
+    private val championshipRepository: ChampionshipDAO,
+    private val userRepository: UserDAO,
+    private val userByChampionshipRepository: UserByChampionshipDAO
+) {
+
+    fun create(request: CreateChampionshipRequest): Pair<Long, String> {
+        val ownerUser = userRepository.findByUsername(request.owner).first()
+        val newChampionship = championshipRepository.save(createEntity(request, ownerUser))
+        saveInUserByChampionship(ownerUser.username, newChampionship)
+        return Pair(newChampionship.idChampionship, newChampionship.name)
+    }
+
+    fun addUser(championshipId: Long, username: String) {
+        val foundChampionships =
+            championshipRepository.findByIdChampionship(championshipId)
+        val user = userRepository.findByUsername(username)
+        if (user.isEmpty()) {
+            throw UnknownUserException("User with username $username not found")
+        }
+        return if (foundChampionships.isNotEmpty()) {
+            val championship = foundChampionships.first()
+            saveInUserByChampionship(username, championship)
+        } else {
+            throw ChampionshipNotFoundException(championshipId)
+        }
+    }
```

Pondría un if para `foundChamionships.isEmpty()` y ahí tirar la excepción, para fail fast